web/test/util.test.js

```diff
@@ -21,3 +21,45 @@
   });
```

Comment:
This is a good suggestion.

Reply:
> Also test that the URL parameters are properly encoded, especially for scopes with special characters.

After changing to https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams this is not necessarily a testing issue on our end. Should I add something quick and dirty in case the API changes?

> the case when query_params contains values that would override the default params

This should no longer be possible.

> Consider adding tests for: ... invalid client_id format

Should I bring in UUID types and [validation](https://www.npmjs.com/package/uuid#uuidvalidatestr), @JoshuaSBrown @nedvedba ? 

> Consider adding tests for: ... empty scopes array, undefined/null parameters

Is there a quick and idiomatic way to test for undefined/null parameters in JavaScript? In my opinion, this is part of the contract established in both the JSDoc and the function signature, but I know that invalid parameters *could* be provided anyway.

Reply:
I have added default parameters where they may have been ambiguous before, and pulled defaults from the reference material.